auth-api/src/test/java/io/stargate/auth/api/AuthResourceTest.java

```diff
@@ -0,0 +1,260 @@
+package io.stargate.auth.api;
+
+import io.dropwizard.testing.junit5.DropwizardExtensionsSupport;
+import io.dropwizard.testing.junit5.ResourceExtension;
+import io.stargate.auth.AuthenticationService;
+import io.stargate.auth.UnauthorizedException;
+import io.stargate.auth.model.AuthTokenResponse;
+import io.stargate.auth.model.Credentials;
+import io.stargate.auth.model.Secret;
+import io.stargate.auth.model.UsernameCredentials;
+import io.stargate.auth.model.Error;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(DropwizardExtensionsSupport.class)
+class AuthResourceTest {
+
+    private static final AuthenticationService authService = mock(AuthenticationService.class);
+
+    private static final ResourceExtension resourceWithUsernameTokenDisabled = ResourceExtension.builder()
+            .addResource(new AuthResource(authService, false))
+            .build();
+
+    private static final ResourceExtension resourceWithUsernameTokenEnabled = ResourceExtension.builder()
+            .addResource(new AuthResource(authService, true))
+            .build();
+
+    @AfterEach
+    void tearDown() {
+        reset(authService);
+    }
+
+    @Test
+    void createTokenFromSecretSuccess() throws UnauthorizedException {
+        Secret secret = new Secret("key", "secret");
+        when(authService.createToken("key", "secret")).thenReturn("token");
+
+        AuthTokenResponse token = resourceWithUsernameTokenDisabled.target("/v1/auth/token/generate")
+                .request()
+                .post(Entity.entity(secret, MediaType.APPLICATION_JSON), AuthTokenResponse.class);
+
+        assertThat(token.getAuthToken()).isEqualTo("token");
+    }
+
+    @Test
+    void createTokenFromSecretUnauthorized() throws UnauthorizedException {
+        Secret secret = new Secret("key", "secret");
+        when(authService.createToken("key", "secret")).thenThrow(UnauthorizedException.class);
+
+        Response response = resourceWithUsernameTokenDisabled.target("/v1/auth/token/generate")
+                .request()
+                .post(Entity.entity(secret, MediaType.APPLICATION_JSON));
+
+        assertThat(response.getStatusInfo().getStatusCode()).isEqualTo(401);
+        assertThat(response.readEntity(Error.class).getDescription()).isEqualTo("Failed to create token: null");
```

Comment:
The error message does not look very nice to me in this context. I'd expect something like `Not authorized`... certainly the `null` part is not user-friendly... but this is probably beyond the scope of this PR.

Reply:
Well the null is do to the mock [up above](https://github.com/stargate/stargate/pull/248/files/9404ac7ea0abcc8afade7401d9f39082eb11d4f2#diff-60c16ee2dabf251021b91943dec2519ab1af6f379e0ff58e733b582070550ef2R58). Running for real this would have the error from C*

Reply:
Ah, in this case, could we change the mock to use some message that could be later validated here?

Reply:
Added some more meaningful dummy messages.